src/util/BitSet.h

```diff
@@ -0,0 +1,252 @@
+#pragma once
+
+// Copyright 2019 Stellar Development Foundation and contributors. Licensed
+// under the Apache License, Version 2.0. See the COPYING file at the root
+// of this distribution or at http://www.apache.org/licenses/LICENSE-2.0
+
+// C++ value-semantic / convenience wrapper around C bitset_t
+
+#include <memory>
+#include <ostream>
+#include <set>
+
+extern "C" {
+#include "cbitset.h"
+};
+
+class BitSet
+{
+    mutable bool mCountDirty = {true};
+    mutable size_t mCount = {0};
+    std::unique_ptr<bitset_t, decltype(&bitset_free)> mPtr;
+
+  public:
+    BitSet() : mPtr(bitset_create(), &bitset_free)
+    {
+    }
+    BitSet(size_t n) : mPtr(bitset_create_with_capacity(n), &bitset_free)
+    {
+    }
+    BitSet(std::set<size_t> const& s)
+        : mPtr(bitset_create_with_capacity(s.empty() ? 0 : *s.end()),
```

Comment:
are you dereferencing s.end() here or I'm reading this wrong?

Reply:
Oh! It appears I am! I don't know what I was thinking; it should be *s.rbegin(). Fixed, thanks!

Reply:
I thought this would crash with `--enable-extra-checks` ... oh except they don't work on Mac?

Reply:
I just don't always build with them on as I'm often doing perf work. switched to that now.